docs/src/main/asciidoc/amazon-lambda.adoc

```diff
@@ -240,33 +294,469 @@ of things you need to do.  Take a look at the generated example project to get a
+            </dependency>
+
+        </dependencies>
+    </dependencyManagement>
+    <dependencies>
+
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>url-connection-client</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>apache-client</artifactId>
+            <exclusions>
+                <exclusion>
+                    <groupId>commons-logging</groupId>
+                    <artifactId>commons-logging</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <!-- sqs/sns/s3 etc -->
+            <artifactId>sqs</artifactId>
+            <exclusions>
+                <!-- exclude the apache-client and netty client -->
+                <exclusion>
+                    <groupId>software.amazon.awssdk</groupId>
+                    <artifactId>apache-client</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>software.amazon.awssdk</groupId>
+                    <artifactId>netty-nio-client</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>commons-logging</groupId>
+                    <artifactId>commons-logging</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jboss.logging</groupId>
+            <artifactId>commons-logging-jboss-logging</artifactId>
+            <version>1.0.0.Final</version>
+        </dependency>
+    </dependencies>
+----
+
+NOTE: if you see `java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty` or similar SSL error, due to the current status of GraalVM,
+there is some additional work to bundle the function.zip, as below.  For more information, please see the link:native-and-ssl[Quarkus Native SSL Guide].
+
+NOTE: If you have a Linux installation of GraalVM, you can reference the artifacts directly; if not, you will need to use Docker.
+
+To bundle the missing artifacts, we need to first extract them from the Docker image used to build the native image.
+
+We do this by starting up a Docker container, in the background, and attaching to that container to copy the artifacts.
+
+First, let's start the GraalVM container, noting the container id output.
+[source, shell]
+----
+docker run -it -d --entrypoint bash quay.io/quarkus/ubi-quarkus-native-image:19.3.1-java8
+
+# This will output a container id, like 6304eea6179522aff69acb38eca90bedfd4b970a5475aa37ccda3585bc2abdde
+# Note this value as we will need it for the commands below
+----
+
+First, libsunec.so, the C library used for the SSL implementation:
+
+[source]
+----
+docker cp {container-id-from-above}:/opt/graalvm/jre/lib/amd64/libsunec.so src/main/resources/
```

Comment:
This is not needed anymore with GraalVM 19.3. As it's the minimal version now, we can get rid if that sunec thing.

Reply:
Unfortunately that's not the case, I wish it was.  GraalVM 19.3.1 or otherwise.  I've tried Quarkus 1.2.0.Final, 1.2.1.Final, 1.3.0.Final and 999-SNAPSHOT (HEAD) and it's all the same.  

For example see #7846.

For anyone going through that pain (which can turn them away from Quarkus & native-image), I think there's value in having the info there, until it never happens again.  Happy to also publish a reproducible project to show this, it's how I've generated the detail for the documentation featured here.

With 1.3.0.Final and GraalVM 19.3.1 (and native-image SSL enabled):


```
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on GraalVM Version 19.3.1 CE

286 | [INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] /opt/graalvm/bin/native-image -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Dcom.sun.xml.bind.v2.bytecode.ClassTailor.noOptimize=true -J-Duser.language=en -J-Dfile.encoding=ANSI_X3.4-1968 --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -H:+JNI -jar gw-sqs-push-1.0-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:-IncludeAllTimeZones -H:EnableURLProtocols=http,https --enable-all-security-services -H:NativeLinkerOption=-no-pie --no-server -H:-UseServiceLoaderFeature -H:+StackTrace gw-sqs-push-1.0-SNAPSHOT-runner

```


Invoking function (sh manage.sh native invoke)

```
483 | ++ aws lambda invoke response.txt --function-name GwSqsPushNativeFunction --payload file://payload.json --log-type Tail --query LogResult --output text
484 | ++ base64 -d
485 | START RequestId: ea42071e-111e-4d19-b1ee-3da17e54d7cf Version: $LATEST
486 | handleRequest: Invoked with request [ SQSRequest(firstName=Timothy, lastName=Power) ]
487 | 2020-03-19 05:25:42,649 INFO  [org.tjp.LambdaSQSPush] (Lambda Thread) handleRequest: Invoked with request [ SQSRequest(firstName=Timothy, lastName=Power) ]
488 |  
489 | handleRequest: Error [ Unable to execute HTTP request: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty ]
490 | [Ljava.lang.StackTraceElement;@7fd3712e4c98END RequestId: ea42071e-111e-4d19-b1ee-3da17e54d7cf
491 | REPORT RequestId: ea42071e-111e-4d19-b1ee-3da17e54d7cf  Duration: 428.63 ms Billed Duration: 500 ms Memory Size: 256 MB Max Memory Used: 40 MB  Init Duration: 18.13 ms
492 | {"body":"{ \"exception\": \"Unable to execute HTTP request: java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty\" }","headers":{"Content-Type":"application/json","X-Custom-Header":"application/json"},"statusCode":500}/codebuild/output/src105077389/src

```

Reply:
Marking this as resolved, since it's still an issue, is that ok?